The new Chat System: surf-social

TheBjoRedCraft opened the pull request:
### 🇩🇪 **Beschreibung – Minecraft Chat System Plugin**  
Das **Chat System Plugin** erweitert den Minecraft-Chat mit neuen Funktionen wie **Direktnachrichten**, **individuelle Chat-Formate**, **Ignorier-Funktion** und die Möglichkeit, **DMs zu deaktivieren**. Perfekt für Server, die mehr Kontrolle über den Chat möchten!  

#### 🚀 Funktionen:  
✅ **Direktnachrichten (DMs):** Flüstere anderen Spielern privat zu.  
✅ **Chat-Formatierung:** Passe das Chat-Layout nach deinen Wünschen an.  
✅ **Ignorier-Funktion:** Blockiere Spieler, um ihre Nachrichten nicht mehr zu sehen.  
✅ **DMs an/aus:** Bestimme, ob du private Nachrichten empfangen möchtest.  

#### 🛠️ Befehle:  
📩 `/msg <Spieler> <Nachricht>` – Sende eine private Nachricht.  
🚫 `/ignore <Spieler>` – Blende Nachrichten eines Spielers aus.  
🔇 `/toggleDMs` – Aktiviere oder deaktiviere Direktnachrichten.  

Funfact: Mit kotlin!

TheBjoRedCraft committed:
feat(surf-chat): add buildgradlekts

TheBjoRedCraft committed:
feat(surf-chat): add basicfriendapi

TheBjoRedCraft committed:
feat(surf-chat): add hikaricp

TheBjoRedCraft committed:
feat(surf-chat): add todo message

TheBjoRedCraft committed:
feat(surf-chat): refactor renaming "message" to "send"

TheBjoRedCraft committed:
feat(surf-chat): pm toggle cmd

TheBjoRedCraft committed:
feat(surf-chat): ignore cmd

TheBjoRedCraft committed:
feat(surf-chat): add every sended message to history

TheBjoRedCraft committed:
feat(surf-chat): remove config provider

TheBjoRedCraft committed:
feat(surf-chat): add chatuser + database

TheBjoRedCraft committed:
feat(surf-chat): add database config

TheBjoRedCraft committed:
feat(surf-chat): chek if player has toggled

TheBjoRedCraft committed:
feat(surf-chat): add minimessage builder part

TheBjoRedCraft committed:
feat(surf-chat): working on replacing with correct system

TheBjoRedCraft committed:
feat(surf-chat): fix PlayerAsyncChatListener and use new message system

TheBjoRedCraft committed:
feat(surf-chat): add component (Component#append(Component)) to messagebuilder

TheBjoRedCraft committed:
feat(surf-chat): add muted check

TheBjoRedCraft committed:
feat(surf-chat): save user when removing/invalidating

TheBjoRedCraft committed:
feat(surf-chat): use jdk21

TheBjoRedCraft committed:
feat(surf-chat): change color

TheBjoRedCraft committed:
feat(surf-chat): add white to builder

TheBjoRedCraft committed:
feat(surf-chat): fix message deletion?

TheBjoRedCraft committed:
feat(surf-chat): register commands [last commit before kotlin]

TheBjoRedCraft committed:
Rename .java to .kt

TheBjoRedCraft committed:
feat(surf-chat): kotlin

TheBjoRedCraft committed:
feat(surf-chat): cleanup and fix commands

TheBjoRedCraft committed:
feat(surf-chat): cleanup and fix listener

TheBjoRedCraft committed:
feat(surf-chat): cleanup and fix objects

TheBjoRedCraft committed:
feat(surf-chat): cleanup and fix external

TheBjoRedCraft committed:
feat(surf-chat): cleanup and fix provider

TheBjoRedCraft committed:
feat(surf-chat): services

TheBjoRedCraft committed:
feat(surf-chat): cleanup and fix main class

TheBjoRedCraft committed:
feat(surf-chat): cleanup and fix

TheBjoRedCraft committed:
feat(surf-chat): fix compiling errors

TheBjoRedCraft committed:
feat(surf-chat): add async stuff

TheBjoRedCraft committed:
feat(surf-chat): add save cmd

TheBjoRedCraft committed:
feat(surf-chat): set java to 21 for mc 1.21.4

TheBjoRedCraft committed:
feat(surf-chat): fix name

TheBjoRedCraft committed:
feat(surf-chat): add more checks

TheBjoRedCraft committed:
feat(surf-chat): add more checks

TheBjoRedCraft committed:
feat(surf-chat): add more checks

TheBjoRedCraft committed:
feat(surf-chat): change message

TheBjoRedCraft committed:
feat(surf-chat): add message

TheBjoRedCraft committed:
feat(surf-chat): change message handling when ignoring

TheBjoRedCraft committed:
feat(surf-chat): dont show up message when player is ignored

TheBjoRedCraft committed:
feat(surf-chat): save all users on shutdown

TheBjoRedCraft committed:
feat(surf-chat): add click-to-answer

TheBjoRedCraft committed:
feat(surf-chat): fix message remove bug

TheBjoRedCraft committed:
feat(surf-chat): idea: velocity (bukkit side) messagesending

TheBjoRedCraft committed:
feat(surf-chat): use Components.kt

TheBjoRedCraft committed:
feat(surf-chat): use object class types

TheBjoRedCraft committed:
feat(surf-chat): fix errors while using objects

TheBjoRedCraft committed:
feat(surf-chat): dont use !!

TheBjoRedCraft committed:
feat(surf-chat): add deprecation for VelocityProxyService.kt

TheBjoRedCraft commented:
Bereit zum checken @ammodev @twisti-dev

TheBjoRedCraft committed:
feat(surf-chat): .

TheBjoRedCraft committed:
feat(surf-chat): add also check spam etc when private msg

TheBjoRedCraft committed:
feat(surf-chat): reply command

TheBjoRedCraft committed:
feat(surf-chat): invalidate on quit

TheBjoRedCraft committed:
feat(surf-chat): use paper renderer

TheBjoRedCraft commented:
@ammodev @twisti-dev

twisti-dev commented:
Kannst du auf das surf-api gradle plugin umstellen? https://github.com/SLNE-Development/surf-api/tree/version/1.21/surf-api-gradle-plugin

twisti-dev commented:
Und dann auch erstmal die commands auf kotlin machen? CommandAPI hat kotlin extensions, die sind dann in surf-api drin https://commandapi.jorel.dev/9.7.0/kotlinintro.html

TheBjoRedCraft committed:
feat(surf-chat): add commandapi kotlin dependencys

TheBjoRedCraft committed:
feat(surf-chat): update commands to use cmdapikotlin

TheBjoRedCraft committed:
feat(surf-chat): add surf api gradle plugin

TheBjoRedCraft committed:
feat(surf-chat): fix building

TheBjoRedCraft commented:
> Und dann auch erstmal die commands auf kotlin machen? CommandAPI hat kotlin extensions, die sind dann in surf-api drin https://commandapi.jorel.dev/9.7.0/kotlinintro.html

erledigt

TheBjoRedCraft commented:
> Kannst du auf das surf-api gradle plugin umstellen? https://github.com/SLNE-Development/surf-api/tree/version/1.21/surf-api-gradle-plugin

erledigt

twisti-dev commented on surf-chat/build.gradle.kts:
```diff
@@ -0,0 +1,89 @@
+}
+
+dependencies {
+    compileOnly("io.papermc.paper:paper-api:1.21.4-R0.1-SNAPSHOT")
```
paper-api
commandapi
placeholder api
caffeine
mccouroutine (folia)
kotlin + coroutines

ist in surf-api drin

TheBjoRedCraft replied:
sicher das papi drin ist?

TheBjoRedCraft replied:
alles außer papi erledigt

twisti-dev commented on surf-chat/build.gradle.kts:
```diff
@@ -0,0 +1,89 @@
+    authors.add("SLNE Development")
+
+    serverDependencies {
+        registerRequired("CommandAPI")
```
kommt durch surf-api

TheBjoRedCraft replied:
ist papi auch als dependency drin?

TheBjoRedCraft replied:
erledigt außer papi

twisti-dev replied:
Ja

twisti-dev replied:
achne nur für placeholder

twisti-dev replied:
Machst du nur placeholder oder parsed du die auch?

TheBjoRedCraft replied:
beim chat dings replacen weil die luckperms api kacke ist

twisti-dev commented on surf-chat/build.gradle.kts:
```diff
@@ -0,0 +1,89 @@
+import dev.slne.surf.surfapi.gradle.util.registerRequired
+
+plugins {
+    id("java")
```
java, shadow, kotlin jvm ist auch in surf-api

TheBjoRedCraft replied:
erledigt

twisti-dev commented on surf-chat/src/main/java/dev/slne/surf/social/chat/listener/PlayerAsyncChatListener.kt:
```diff
@@ -0,0 +1,108 @@
+        val player = event.player
+        var plainMessage = PlainTextComponentSerializer.plainText().serialize(event.message())
+
+        if (event.isCancelled) {
```
Das event sollte garnicht aufgerufen werden außer du hast die priority auf MONITOR oder ignoreCancelled auf true

TheBjoRedCraft replied:
erledigt

twisti-dev commented on surf-chat/src/main/java/dev/slne/surf/social/chat/listener/PlayerAsyncChatListener.kt:
```diff
@@ -0,0 +1,108 @@
+import java.security.SecureRandom
+
+class PlayerAsyncChatListener : Listener {
+    private val random: SecureRandom = SecureRandom()
```
du kannst einfach das random aus surf-api benutzen 
das ist eine public val die auch random heißt

TheBjoRedCraft replied:
erledigt (egal durch das hier drunter)

twisti-dev commented on surf-chat/src/main/java/dev/slne/surf/social/chat/listener/PlayerAsyncChatListener.kt:
```diff
@@ -0,0 +1,108 @@
+        }
+
+        val channel: Channel? = Channel.getChannel(player)
+        val messageID: Int = random.nextInt(1000000)
```
wäre eine uuid nicht besser? Hier kann es schon mal vorkommen das zweimal die gleiche id generiert wird

TheBjoRedCraft replied:
erledigt

twisti-dev commented on surf-chat/src/main/java/dev/slne/surf/social/chat/util/MessageBuilder.kt:
```diff
@@ -0,0 +1,162 @@
+import net.kyori.adventure.text.event.HoverEvent
+import net.kyori.adventure.text.minimessage.MiniMessage
+
+class MessageBuilder {
```
Brauchst du den? Surf api hat einen component builder (buildText {})

TheBjoRedCraft replied:
meiner ist besser ⭐⭐⭐

TheBjoRedCraft replied:
würde mein erstmal lassen, wenns passt

twisti-dev committed:
feat(channel): replace ObjectArraySet with mutableObjectSetOf for collections

TheBjoRedCraft committed:
feat(surf-chat): "kommt durch surf-api"

TheBjoRedCraft committed:
feat(surf-chat): paper-api
                 commandapi
                 caffeine
                 mccouroutine (folia)
                 kotlin + coroutines

                 ist in surf-api drin

TheBjoRedCraft committed:
feat(surf-chat): java, shadow, kotlin jvm ist auch in surf-api

TheBjoRedCraft committed:
feat(surf-chat): Das event sollte garnicht aufgerufen werden außer du hast die priority auf MONITOR oder ignoreCancelled auf true

TheBjoRedCraft committed:
feat(surf-chat): du kannst einfach das random aus surf-api benutzen
                 das ist eine public val die auch random heißt

TheBjoRedCraft committed:
feat(surf-chat): wäre eine uuid nicht besser? Hier kann es schon mal vorkommen das zweimal die gleiche id generiert wird

TheBjoRedCraft committed:
Merge remote-tracking branch 'origin/dev/chat' into dev/chat

twisti-dev committed:
feat(channel): refactor channel member management to use UUIDs and improve online player handling (not tested tho 🤞)

twisti-dev committed:
feat: update color handling to use centralized Colors class for hover events

twisti-dev committed:
feat: refactor cache handling in ChatUser to use new Caffeine API and improve performance

twisti-dev committed:
feat: refactor ChannelProvider and ChatFilterService to use mutable collections for improved performance and readability

twisti-dev committed:
feat: refactor ChatHistoryService to improve cache handling and message broadcasting

twisti-dev committed:
feat: refactor ChatReplyService and DatabaseService for improved performance and readability

twisti-dev committed:
feat: introduce SurfChatPermissions for centralized permission management and refactor component methods for improved readability

twisti-dev committed:
feat: simplify ChannelArgument and ChannelInviteArgument constructors for improved readability

twisti-dev committed:
feat: enhance ignore command with self-mute prevention and improved feedback messages

TheBjoRedCraft committed:
fix(surf-chat): fix building problem?

TheBjoRedCraft committed:
Merge remote-tracking branch 'origin/dev/chat' into dev/chat

TheBjoRedCraft committed:
fix(surf-chat): fix building problem?
